cpp/src/copying/scatter.cu

```diff
@@ -222,6 +225,55 @@ struct column_scalar_scatterer {
   }
 };
 
+template <typename MapIterator>
+struct column_scalar_scatterer_impl<struct_view, MapIterator> {
+  std::unique_ptr<column> operator()(std::reference_wrapper<const scalar> const& source,
+                                     MapIterator scatter_iter,
+                                     size_type scatter_rows,
+                                     column_view const& target,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr) const
+  {
+    // For each field of `source`, copy construct a scalar from the field
+    // and dispatch the correct scalar scatterer
+
+    auto typed_s       = static_cast<struct_scalar const*>(&source.get());
+    size_type n_fields = typed_s->view().num_columns();
+    CUDF_EXPECTS(n_fields == target.num_children(), "Mismatched number of fields.");
+
+    auto scatter_functor = column_scalar_scatterer<decltype(scatter_iter)>{};
+    std::vector<std::unique_ptr<column>> fields;
+    std::transform(thrust::make_counting_iterator(0),
+                   thrust::make_counting_iterator(n_fields),
+                   std::back_inserter(fields),
+                   [&](auto const& i) {
+                     auto row_slr = get_element(typed_s->view().column(i), 0, stream);
```

Comment:
This can be done slightly more declaratively with a `std::vector` range ctor and a `cudf::detail::counting_transform_iterator`